src/geometry/ArborX_HyperPoint.hpp

```diff
@@ -34,15 +36,11 @@ struct Point
   FloatingPoint _coords[DIM] = {};
 };
 
-// Deduction guides
-template <class T>
-Point(T x) -> Point<1, T>;
-
-template <class T>
-Point(T x, T y) -> Point<2, T>;
-
-template <class T>
-Point(T x, T y, T z) -> Point<3, T>;
+template <typename... T>
+Point(T...)
+    -> Point<sizeof...(T), std::conditional_t<
+                               (... || std::is_same_v<std::decay_t<T>, double>),
```

Comment:
How does this work? Which argument will be decayed? First one, or any of them? If we do `(0, 0.2)`, would that still decay to `double`?

Reply:
All of them are cast to `double` if any of the arguments is a double and if not all of them are cast to `float`.
`(0, 0.2)` would give a `Point<2, double>`.

Reply:
The question still is how to get around `Wc++11-narrowing`-type warnings.

Reply:
Ultimately, we can't do the automatic conversion using aggregate initialization so we would need to provide a constructor (or force the user to provide the correct type).

> I really need to learn more about how variadic templates work. Could you please point me to the place where it explains how decay works for a parameter pack? I can't see anything like that [here](https://en.cppreference.com/w/cpp/language/parameter_pack).

I need `decay_t` here since the type might be deduced to `T&&`.

Reply:
```
/tmp/ArborX/test/tstQueryTreeCallbacks.cpp:102:23: error: non-constant-expression cannot be narrowed from type 'double' to 'float' in initializer list [-Wc++11-narrowing]
        points(i) = {{(double)i, (double)i, (double)i}};
                      ^~~~~~~~~
```
using https://github.com/aprokop/ArborX/pull/17 for example.

Reply:
I think that is fine. In the sense, that actually is something a user needs to address.

Reply:
Yes, it's just that this pull request doesn't address all problems in aprokop#17.